crmd/remote_lrmd_ra.c

```diff
@@ -1072,3 +1072,43 @@ remote_ra_fail(const char *node_name)
     }
 }
 
+/* PE passes hash2metafield() equivalent of XML_RSC_ATTR_REMOTE_NODE */
+#define PE_ATTR_REMOTE_NODE "CRM_meta_remote_node"
+
+/* A guest node stop implied by host fencing looks like:
+ *  <pseudo_event id="36" operation="stop" operation_key="vm1_stop_0">
+ *     <attributes CRM_meta_remote_node="guest1" .../>
+ *  </pseudo_event>
+ */
+#define XPATH_PSEUDO_STOP "//" XML_GRAPH_TAG_PSEUDO_EVENT \
+    "[@" XML_LRM_ATTR_TASK "='" CRMD_ACTION_STOP "']/" XML_LRM_TAG_ATTRIBUTES
+
+/*!
+ * \internal
+ * \brief Check a pseudo-action for Pacemaker Remote node side effects
+ *
+ * \param[in] xml  XML of pseudo-action to check
+ */
+void
+remote_ra_pseudo(xmlNode *xml)
+{
+    xmlXPathObjectPtr search = xpath_search(xml, XPATH_PSEUDO_STOP);
+
+    if (numXpathResults(search) == 1) {
+        xmlNode *attr = getXpathResult(search, 0);
+        const char *remote;
+
+        /* Normally, we handle the necessary side effects of a guest node stop
+         * action when reporting the remote agent's result. However, if the stop
+         * is implied due to fencing, it will be a pseudo-event, and there won't
+         * be a result to report. Handle that case here.
+         */
+        if (attr) {
+            remote = crm_element_value(attr, PE_ATTR_REMOTE_NODE);
+            if (remote) {
+                remote_node_down(remote);
```

Comment:
I double checked the `remote_node_down()` function, but it still seems like something needs to happen on the node hosting the guest and/or its connection.